### ISSUE TITLE
First Sprint

### ISSUE THREAD

**Issue by Kahuna915** [Kahuna915/Capstone-Design#21]
# Sprint Statement:
The goal of this Sprint is to get organized. From understanding our client's wants for our project to creating a layout to go forward, and solving any unresolved issues from last semester. Our focus is also on creating effective and innovative solutions for the separation of user environments. 
# Week 1
Week 1, we will need to create a new organization, set up billing, and create organization units before we can start to add accounts. Once the OU's are created to represent cyber.local's hierarchy, we can then add users.
-  [X] [Create Organization](https://github.com/Kahuna915/Capstone-Design/wiki/AWS-Organization-Set-up) | Tanner (As we face various problems, I am putting creating a new organization on hold until we can smoothly transition into a production environment. Our testing will continue in the organization that was created last semester.)
-  [x] [Create OU](https://github.com/Kahuna915/Capstone-Design/wiki/Design-Project-1-Objectives-Sep-15th---Oct-16th) | Tanner
-  [ ] [Set up billing](https://github.com/Kahuna915/Capstone-Design/wiki/Design-Project-1-Objectives-Sep-15th---Oct-16th) | Alex
     -  Credit Card information
     -  Maximum Limit
-  [X] [Fix Separation of users](https://github.com/Kahuna915/Capstone-Cloud-Integration/issues/26) | Noah 
## Kickoff meeting
A meeting will be organized with the client to arrange the name of the new organization, implementation of the credit card, and any other features requested by the client.   
**This will also count as our week 1 progress update**

# Week 2
In week 2, we will need to create users, optimize our budget reports, and create a propagated billing system.
-  [X] [Creating organization users](https://github.com/Kahuna915/Capstone-Design/wiki/Automated-account-creation-using-AWS-CLI) | Noah Script
- [x] [Budget Reports](https://github.com/Kahuna915/Capstone-Design/wiki/Design-Project-2-Objectives-Oct-16th---November-15th) | Alex
- [x] [Propogated Billing](https://github.com/Kahuna915/Capstone-Design/wiki/Design-Project-2-Objectives-Oct-16th---November-15th) | Alex
- [ x] [Separate Users](https://github.com/Kahuna915/Capstone-Design/wiki/Design-Project-2-Objectives-Oct-16th---November-15th) | Noah and Tanner

## Progress update
At the end of week 2, we will need to do a progress update to make sure the team is at a good point.

# Week 3
In Week 3, we will join AWS with our on-prem AD, this will allow us to potentially use federated logins, or copy our policies from AD
- [ ] [Join AD](https://github.com/Kahuna915/Capstone-Design/wiki/AWS-Organization-Set-up) | Whole team

## Progress Update
This week, I found that natively AWS member accounts have independent environments, allowing the separation of users. I have worked on creating SCP policies to prevent users from accessing billing and editing policies for the organization. Admin accounts are also able to view Members' accounts and environments. Next week I will set up Logging for the organization and update policies as necessary. - Tanner

An update following the first sprint, have created a few scripts to automate various tasks, setup basic network connections. Solved separated users...(Security Policies impacting us). Working on AD Connector.. a bit more complicated than I originally thought. - Noah

# Demo
We kept our test environment, but deleted almost everything. Solved separation, automation and other various tasks. Currently working on ad connector test and documentation. Would like to rebuild the Github. 

### Reflection
The sprint went well. We had a bunch of breakthroughs in areas we were struggling. Due to the breakthroughs we have had a pretty good sprint. I believe we solved key issues which were limiting us from continuing. We also changed the path of the project slightly and believe this will help us be more efficient. By starting smaller and then getting bigger at the end.

**Comment by Kahuna915** [comment]
Currently on Week 3 setup the prerequisites for connecting the AD and sent email to Devin on Wednesday the 1st.  Waiting for response, just need the fully qualified domain name, the IP of the DNS servers we are connecting to, a service account username and password.

**Comment by Kahuna915** [comment]
Not restarting the org because it took two weeks for amazon to even spin it up. Have created some more org users for testing, and have everything set for the ad connection just waiting on response from Devin.
